06-building-a-program-start-to-finish/writing-tests/output/spec.js

```diff
@@ -0,0 +1,122 @@
+/* My Test Setups */
+/* global describe it expect */
+import myBook from './books.json';
+import emptyBook from './emptyBook.json';
+import invalidKeys from './Invalidkeys.json';
+
+/* My Suites */
+describe('Class and Method Instantaion', () => {
+  const myClass = new InvertedIndex();
+
+  it('Should be instantiated with the new keyword', () => {
+    const init = () => {
+      InvertedIndex();
+    };
+    expect(init).toThrowError('Cannot call a class as a function');
+  });
+
+  it('Should contain the getIndex method', () => {
+    expect(typeof myClass.getIndex).toBe('function');
+  });
+
+  it('Should contain the createIndex method', () => {
+    expect(typeof myClass.createIndex).toBe('function');
+  });
+
+  it('Should contain the searchIndex method', () => {
+    expect(typeof myClass.searchIndex).toBe('function');
+  });
+
+  it('Should contain the readFile method', () => {
+    expect(typeof myClass.readFile).toBe('function');
+  });
+
+  it('Should contain the tokenize method', () => {
+    expect(typeof InvertedIndex.tokenize).toBe('function');
+  });
+
+  it('Should contain the validateFile method', () => {
+    expect(typeof InvertedIndex.validateFile).toBe('function');
+  });
+});
+
+describe('Populating Data', () => {
+  const myClass = new InvertedIndex();
```

Comment:
I see that you created an instance if your invertedIndex class above. Is there a better way of handling this?